lisa/tests/staging/utilclamp.py

```diff
@@ -0,0 +1,296 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2020, Arm Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import functools
+from operator import itemgetter
+import itertools
+
+import pandas as pd
+import numpy as np
+from devlib.utils.misc import memoized
+
+from lisa.analysis.frequency import FrequencyAnalysis
+from lisa.analysis.load_tracking import LoadTrackingAnalysis
+from lisa.datautils import df_add_delta
+from lisa.datautils import series_mean
+from lisa.pelt import PELT_SCALE
+from lisa.tests.base import ResultBundle, RTATestBundle
+from lisa.wlgen.rta import Periodic
+
+
+class UtilClamp(RTATestBundle):
+    """
+    Validate that UtilClamp min values are honoured properly by the kernel.
+
+    The test is split into 8 phases. For each phase, a UtilClamp value is set
```

Comment:
Adding a new comment since @valschneider 's one was marked as resolved and apparently cannot be resurrected:

> That smells like annoying refactoring, but this feels like something that should be a parameter rather than a hardcoded value.
> Annoyingly this controls the rtapp profile, so this can't be a test parameter... Perhaps a class attribute for now?

We should be able to add an optional parameter to `_from_target()` method like `UserspaceSanity`:
```
    @classmethod
    def _from_target(cls, target: Target, *,
		res_dir: ArtifactPath = None,
		nr_phases=None,
    ) -> 'UtilClamp':
```

And in `__init__(..., nr_phases=None)`, we replace `None` by 8. Hardcoding 8 as default value in different paths leads to missed updates when the value needs to change, it's much better to use a standard `None` "marker" and deal with that in a single place like:
```
self.nr_phases = 8 if nr_phases is None else nr_phases 
```

with the appropriate docstring.